### PR TITLE
collect all clients to one single folder

### PR DESCRIPTION
**Pull request by Sunt-ing**
After reading every line of code in this repo, I think the names of our three clients (`client.rs`, `buffer.rs`, `blocking_client.rs`) can be less confusing for me (and maybe other green hands) if they are unified. 

So in this PR, I re-org the `src/` and `tests/` to make them more clear:
```
├── src
│   ├── bin
│   │   ├── cli.rs
│   │   └── server.rs
│   ├── clients
│   │   ├── blocking_client.rs
│   │   ├── buffered_client.rs
│   │   ├── client.rs
│   │   └── mod.rs
│   ├── cmd/
│   ├── connection.rs
│   ├── db.rs
│   ├── frame.rs
│   ├── lib.rs
│   ├── parse.rs
│   ├── server.rs
│   └── shutdown.rs
└── tests
    ├── buffered_client.rs
    ├── client.rs
    └── server.rs
```

This is just my personal thought, so feel free to disagree with it and close this PR.

### CODE REVIEW

**Comment by Sunt-ing** [comment]
@carllerche @Darksonn Hi, may I ask what do you think?

File: src/clients/client.rs [hunk 1]
```diff
@@ -99,11 +99,11 @@ impl Client {
     ///
     /// Demonstrates basic usage.
     /// ```no_run
-    /// use mini_redis::client;
+    /// use mini_redis::clients;
     ///
     /// #[tokio::main]
     /// async fn main() {
-    ///     let mut client = client::connect("localhost:6379").await.unwrap();
+    ///     let mut client = clients::connect("localhost:6379").await.unwrap();
```
Comment (Darksonn) [inline src/clients/client.rs:106]: Having to write `clients::connect` is a bit awkward.

Comment (Darksonn) [reply src/clients/client.rs:106]: Perhaps it should be a method on the `Client` type instead?

Comment (Sunt-ing) [reply src/clients/client.rs:106]: Agree. Maybe we can have `connect` methods on the `Client`, `BlockingClient`, and `BufferedClient` types?

**Comment by Sunt-ing** [comment]
@Darksonn Hi Alice, I've changed `clients::connect`, `clients::blocking_connect`, `clients::buffer` to `Client::connect`, `BlockingClient::connect`, `BufferedClient::buffer` respectively. Would please review it again? Thanks!

**Comment by Sunt-ing** [comment]
@carllerche Hi, Carl, what do you think?

**Review by Darksonn — APPROVED** [review]
Thanks, this looks fine to me. Would you be able to update [this page](https://tokio.rs/tokio/topics/bridging) as well? It can be found [here](https://github.com/tokio-rs/website/blob/master/content/tokio/topics/bridging.md).

**Comment by Sunt-ing** [comment]
@Darksonn Hi Alice, I've created a PR in `tokio-rs/website` to update the page.

**Pull request merged**